src/backend/distributed/executor/executor.c

```diff
@@ -2427,8 +2458,8 @@ PlacementExecutionReady(TaskPlacementExecution *placementExecution)
 			 * If the connection is idle, wake it up by checking whether
 			 * the connection is writeable.
 			 */
-			connection->waitFlags = WL_SOCKET_READABLE | WL_SOCKET_WRITEABLE;
-			distributedExecution->waitFlagsChanged = true;
+			UpdateConnectionWaitFlags(session, WL_SOCKET_READABLE | WL_SOCKET_WRITEABLE,
+									  true);
```

Comment:
the overwrite parameter is usually true. What if we remove the parameter and use `session->waitFlags | WL_SOCKET_***` in the non-overwrite cases?

Reply:
Do you mean not calling `UpdateConnectionWaitFlags `? 

I think expecting future users to always keep in mind the decency between `connection->waitFlags ` and  `execution->waitFlagsChanged` is hard.  So, I'd still keep this to prevent bugs in the future, but I do not have strong opinions either. So, if you want, I can do that.

Reply:
I mean remove the overwrite flag and in cases where you want to add to the existing flags call `UpdateConnectionWaitFlags(session, session->waitFlags | WL_SOCKET_READABLE)` instead.